### ISSUE TITLE
API Changes Bike Shed

### ISSUE THREAD

**Issue by sprjr** [sprjr/json-proxy-middleware#2]
While copying things over, and in talking with @jamesplease earlier about potentially making some changes that are friendly with another pattern we have available at Netflix, I thought it would be a good time to discuss potential API changes before publishing & consuming this back into apps. Of course, that will cause some migration issues, but I think we might have those as is (for example, in one of my apps I don't have the `headers()` function and that lives inside my json proxy file for Netflix specific things).

The current API looks (expects) something like this:

```
router.get('some/path/to/proxy', jsonProxyMiddleware({
  grpcClient: { 
    getEurekaOptions() {},
    getUrlHost() {},
  },
  log: {
    info() {},
  },
  msg: '', // any string
  headers (req, res) {}, // a function returning an object or an object
});
```

Obviously a simple change we can consider is `msg` should likely be `message` or `logMessage`, but while we're in there - do we want to change anything related to the API around `grpcClient`, or potentially add in the things related to a `urlHost` option we can use _without_ the grpc option?

**Comment by sprjr** [comment]
Where I landed on the API after some thinkering:

```
jsonProxy({
  additionalLogMessage, // string, optional
  headers, // object, or function returning object
  logger, // bunyan | console | { info(), error() } interface
  urlHost, // string, or function returning a string
})
```

I also **removed** the functionality that responds to clients when there are errors. We now just call `next(Error)` so that applications can handle this however they see fit.

I've _mostly_ implemented this in #1 now, along with adding some `T Y P E S C R I P T` to...try that out. Should be able to get a super early version out some time this week.

**Comment by sprjr** [comment]
All set, I believe :D
